tick/optim/model/src/hawkes_fixed_expkern_loglik.cpp

```diff
@@ -351,6 +362,44 @@ double ModelHawkesFixedExpKernLogLik::hessian_norm_dim_i(const ulong i,
   return hess_norm;
 }
 
+void ModelHawkesFixedExpKernLogLik::hessian_i(const ulong i,
+                                              const ArrayDouble &coeffs,
+                                              ArrayDouble &out) {
+  if (!weights_computed) TICK_ERROR("Please compute weights before calling hessian_i");
+
+  const double mu_i = coeffs[i];
+  const ulong start_alpha_i = n_nodes + n_nodes * i;
+  const ArrayDouble alpha_i = view(coeffs, start_alpha_i, start_alpha_i + n_nodes);
+
+  const ulong start_mu_line = i * (n_nodes + 1);
+  const ulong block_start = (i + 1) * n_nodes * (n_nodes + 1);
+
+  for (ulong k = 0; k < (*n_jumps_per_node)[i]; ++k) {
+    const ArrayDouble g_i_k = view_row(g[i], k);
+
+    double s = mu_i;
+    s += alpha_i.dot(g_i_k);
+    const double s_2 = s * s;
+
+    // fill mu mu
+    out[start_mu_line] += 1. / s_2;
+    // fill mu alpha
+    for (ulong j = 0; j < n_nodes; ++j) {
+      out[start_mu_line + j + 1] += g_i_k[j] / s_2;
```

Comment:
There is a trick here, instead of dividing by `s_2` many times (this line and two below), we can compute` 1.0 / s_2` once and multiply it instead.